src/proposals/20250918/SparkEthereum_20250918.t.sol

```diff
@@ -0,0 +1,249 @@
+
+    constructor() {
+        id = "20250918";
+    }
+
+    function setUp() public {
+        setupDomains("2025-09-10T08:36:00Z");
+
+        deployPayloads();
+
+        // chainData[ChainIdUtils.Ethereum()].payload = 0xe7782847eF825FF37662Ef2F426f2D8c5D904121;
+    }
+
+    function test_ETHEREUM_controllerUpgrade() public onChain(ChainIdUtils.Ethereum()) {
+        _testControllerUpgrade(Ethereum.ALM_CONTROLLER, NEW_ALM_CONTROLLER_ETHEREUM);
+    }
+
+    function test_ETHEREUM_cctpRateLimits() public onChain(ChainIdUtils.Ethereum()) {
+        MainnetController controller = MainnetController(NEW_ALM_CONTROLLER_ETHEREUM);
+
+        _assertUnlimitedRateLimit(controller.LIMIT_USDC_TO_CCTP());
+        _assertRateLimit(
+            RateLimitHelpers.makeDomainKey(controller.LIMIT_USDC_TO_DOMAIN(), CCTPForwarder.DOMAIN_ID_CIRCLE_BASE),
+            50_000_000e6,
+            25_000_000e6 / uint256(1 days)
+        );
+
+        executeAllPayloadsAndBridges();
+
+        _assertUnlimitedRateLimit(controller.LIMIT_USDC_TO_CCTP());
+        _assertRateLimit(
+            RateLimitHelpers.makeDomainKey(controller.LIMIT_USDC_TO_DOMAIN(), CCTPForwarder.DOMAIN_ID_CIRCLE_BASE),
+            200_000_000e6,
+            500_000_000e6 / uint256(1 days)
+        );
+    }
+
+    function test_ETHEREUM_sll_onboardUsdsSpkFarm() public onChain(ChainIdUtils.Ethereum()) {
+        SparkLiquidityLayerContext memory ctx = _getSparkLiquidityLayerContext();
+        MainnetController controller = MainnetController(NEW_ALM_CONTROLLER_ETHEREUM);
+
+        bytes32 depositKey = RateLimitHelpers.makeAssetKey(
+            MainnetController(NEW_ALM_CONTROLLER_ETHEREUM).LIMIT_FARM_DEPOSIT(),
+            USDS_SPK_FARM
+        );
+        bytes32 withdrawKey = RateLimitHelpers.makeAssetKey(
+            MainnetController(NEW_ALM_CONTROLLER_ETHEREUM).LIMIT_FARM_WITHDRAW(),
+            USDS_SPK_FARM
+        );
+
+        IERC20 underlying = IERC20(Ethereum.USDS);
+
+        uint256 expectedDepositAmount = 1_000_000e18;
+
+        deal(address(underlying), address(ctx.proxy), expectedDepositAmount);
+
+        _assertRateLimit(depositKey,  0, 0);
+        _assertRateLimit(withdrawKey, 0, 0);
+
+        executeAllPayloadsAndBridges();
+
+        _assertRateLimit(depositKey,  250_000_000e18,    50_000_000e18 / uint256(1 days));
+        _assertRateLimit(withdrawKey, type(uint256).max, 0);
+
+        assertEq(IFarmLike(USDS_SPK_FARM).balanceOf(address(ctx.proxy)), 0);
+
+        vm.prank(ctx.relayer);
+        controller.depositToFarm(USDS_SPK_FARM, expectedDepositAmount);
+
+        assertEq(ctx.rateLimits.getCurrentRateLimit(depositKey),  250_000_000e18 - expectedDepositAmount);
+        assertEq(ctx.rateLimits.getCurrentRateLimit(withdrawKey), type(uint256).max);
+
+        assertEq(IFarmLike(USDS_SPK_FARM).balanceOf(address(ctx.proxy)), expectedDepositAmount);
+        assertEq(underlying.balanceOf(address(ctx.proxy)),               0);
```

Comment:
Warp and assert that we get a balance of SPK as well when we withdraw. Add SPK before assertions as well.